userspace/libscap/linux/scap_procs.c

```diff
@@ -787,13 +786,25 @@ static int32_t scap_proc_add_from_proc(struct scap_linux_platform* linux_platfor
 
 	// Container start time for host processes will be equal to when the
 	// host init started
-	char proc_cmdline[SCAP_MAX_PATH_SIZE];
-	snprintf(proc_cmdline, sizeof(proc_cmdline), "%scmdline", dir_name);
-	if(stat(proc_cmdline, &dirstat) == 0)
+	char proc_stat[SCAP_MAX_PATH_SIZE];
+	snprintf(proc_stat, sizeof(proc_stat), "%sstat", dir_name);
+	f = fopen(proc_stat, "r");
+	if(f != NULL)
 	{
-		tinfo.clone_ts = dirstat.st_ctim.tv_sec * SECOND_TO_NS + dirstat.st_ctim.tv_nsec;
+		uint64_t start_time_since_boot_ticks;
+		if (fscanf(f, "%*d %*s %*c %*d %*d %*d %*d %*d %*u %*u %*u %*u %*u %*u %*u %*u %*u %*d %*d %*d %*u %lu", &start_time_since_boot_ticks) == 1)
+		{
+			uint64_t start_time_since_boot_ns =
+			           (start_time_since_boot_ticks * SECOND_TO_NS) /
+				    linux_platform->m_generic.m_machine_info.hz;
+			tinfo.clone_ts =
+			    linux_platform->m_generic.m_machine_info.boot_ts_epoch +
```

Comment:
We had a few lengthy discussions about the precision of the boot start ts and timestamps in general in the past. 

boot_ts_epoch is now retrieved by scraping `btime` from `/proc/stat`, but it's not very precise. For the driver we now use `scap_get_precise_boot_time`. Initially we wanted some more constant boot ts, but I would be ok to abandon this after all and go for a boot_ts_epoch that is more precise, but may have super small numerical imprecision when restarting the agent on the same host.

Or we can check out `scap_agent_info`. There are more times mainly used for the metrics feature and CPU usage calculations. Not denying that time is hard as we all know ...

CC @gnosek may have some estimations around perf impact and other ideas.

Reply:
I understand and accept the reluctance to proceed with the approach that I proposed.
I am withdrawing this PR.